### PR TITLE
VIM-5636: Update `VIMUser` Object and Add New Error Codes

### PR DESCRIPTION
**Pull request by supperspidey**
#### Ticket

[VIM-5636](https://vimean.atlassian.net/browse/VIM-5636)

- *Note: this is required for Vimeo staff only. If not applicable to your PR, use "N/A".*

#### Pull Request Checklist

- [x] Resolved any merge conflicts
- [x] No build errors or warnings are introduced
- [x] New files are written in Swift
- [x] New classes contain license headers
- [x] New classes have Documentation
- [x] New public methods have Documentation

#### Issue Summary

The goal of this PR is to add new error codes for live streaming and update the `VIMUser` class to recognize the new `live_quota` response.

#### Implementation Summary

What has been changed:
- `ErrorCode.swift`: added new cases to `VimeoErrorCode` enum
- `VIMUser.h`: added a new property of type `VIMLiveQuota` called `liveQuota`; this property represents the `live_quota` field in the `user` response.
- `VIMLiveTests.swift`: refactored out the `TestingUtility` code into a separate file; modified the code so that it could generically parse any `VIM` object

What has been added:
- `VIMLiveQuota.swift`, `VIMLiveStreams.swift`, `VIMLiveTime.swift`: new classes used to describe the `live_quota` response
- `VIMLiveQuotaTests.swift`: unit tests for parsing the `live_quota` response
- `user_live.json`: test JSON file containing a `user` response with a `live_quota` field

#### Reviewer Tips

N/A

#### How to Test

Make sure all unit tests pass.

### CODE REVIEW

File: VimeoNetworkingExample-iOS/VimeoNetworkingExample-iOSTests/TestingUtility.swift [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import Foundation
+import VimeoNetworking
+
+class TestingUtility<T: AnyObject>
```
Comment (mikew-personal) [inline VimeoNetworkingExample-iOS/VimeoNetworkingExample-iOSTests/TestingUtility.swift]: I feel like I've seen this same code written multiple times and used in networking tests.  You should check out the tests that use OHTTPMock to see how we can use that framework to return JSON data captured in a file.  It's closer to simulating our network calls and tests more of the mapping logic.

File: VimeoNetworking/Sources/Models/VIMUser.m [hunk 2]
```diff
@@ -33,6 +33,7 @@
 #import "VIMPreference.h"
 #import "VIMUploadQuota.h"
 #import "VIMUserBadge.h"
+#import <VimeoNetworking/VimeoNetworking-Swift.h>
```
Comment (mikew-personal) [inline VimeoNetworking/Sources/Models/VIMUser.m]: do we need this import here?  I feel like this is already being included for the whole framework somewhere.  Maybe in the pch?

Comment (supperspidey) [reply VimeoNetworking/Sources/Models/VIMUser.m]: @mikew-personal I don't think so. I checked the `VIMVideo.m`, and that file needed this import to work properly with Swift classes. Should we create an `pch` file for `VimeoNetworking` framework?

**Comment by supperspidey** [comment]
@mikew-personal The PR is ready for another look.

**Comment by cameo-github** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoNetworking/blob/8781d334e3b833d3440afd173b17573fdd69e793/VimeoNetworking/Sources/VimeoClient.swift#L330">VimeoNetworking/Sources/VimeoClient.swift#L330</a></strong>: cast from ‘Response&lt;VIMNullResponse&gt;’ to unrelated type ‘Response&lt;ModelType&gt;’ always fails  <br /><code>
                let response = Response(model: nullResponseObject, json: [:]) as! Response&lt;ModelType&gt;
</code></td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 61 tests, with 0 failures (0 unexpected) in 0.695 (0.726) seconds</td>
    </tr>
  </tbody>
</table>

See build details on [CircleCI](https://circleci.com/gh/vimeo/VimeoNetworking/481)

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: VimeoNetworking/Sources/Models/VIMUser.m (hunk 2 above)
Comment (mikew-personal) [reply VimeoNetworking/Sources/Models/VIMUser.m]: ok, if we're doing the same thing in VIMVideo already then I think this is fine.

**Pull request merged**